langserver/workspace_refs.go

```diff
@@ -100,19 +111,29 @@ func (h *LangHandler) handleWorkspaceReferences(ctx context.Context, conn JSONRP
 					return
 				}
 			}()
+
 			err := h.workspaceRefsFromPkg(ctx, bctx, conn, params, fset, pkg, rootPath, &results)
 			if err != nil {
 				log.Printf("workspaceRefsFromPkg: %v: %v", pkg, err)
 			}
-		}(pkg)
+		}()
+	}
+
+	_, err := h.workspaceRefsTypecheck(ctx, bctx, conn, fset, pkgs, afterTypeCheck)
+	if err != nil {
+		return nil, err
+	}
+
+	// Wait for all worker goroutines to complete.
+	for i := 0; i < int(started); i++ {
```

Comment:
technically you should read started via the atomic pkg. I do realise though that when we get here, there should be no concurrent accesses going on.

Also it seems your started + finishSignal could be accomplished with just a `sync.WaitGroup`